### PR TITLE
Read clio.toml when building project to get hints on build location

### LINKED ISSUE #87: Build command should consider clio.toml to decide where to put bundle
**Is your feature request related to a problem? Please describe.**
`clio build` is sending the compiled files to `.clio/target/node` by default. 

**Describe the solution you'd like**
Not considering the `--destination` parameter override, the output directory for the build command should be configured from `clio.toml`. An example toml config could look like this:

```toml
[build]
directory = "build"
target = "node"

[target.node]
directory = "src"
target = "node"

[target.browser]
directory = "src"
target = "browser"
```

**Describe alternatives you've considered**
We should create two fields on `clio.toml`:

- `target`: to give a hint to the compiler for what target is this build;
- `build`: as discussed on Telegram, to store in which directory to put the builds (default to `build`).

The resulting compiled JavaScript files should be placed in `${build}/${target}`.

**Additional context**
The build command should have a `--target` override to build a specific directory to a specific target. This way, the programmer can generate a project for browser target, but can compile a host for node target using this feature.

### ToDo

- [ ]  Add fields to `packageConfig`
- [ ] Update template repositories
- [ ] Let command line flag override toml

### PR DESCRIPTION
**Pull request by andreyluiz**
Closes #87.

This PR introduces the capability of `clio build` to read project's `clio.toml` searching for a section like this one:
```toml
[build]
directory = "build"
target = "node"

[target.node]
directory = "src"
target = "node"

[target.browser]
directory = "src"
target = "browser"
```
The PR also introduces a new argument on `clio build` to override the target.

Before performing the build, some checks are made to ensure a minimal configuration, or override, is provided to build the destination directory.

If the checks are fulfilled, the build is then placed in `<build.destination>/<build.target>`.

### ToDo
- [x] Add fields to packageConfig
- [x] Update template repositories
- [x] Let command line flag override toml
- [x] `clio build <target>` builds target specified in toml

### CODE REVIEW

File: cli/commands/build.js [hunk 1]
```diff
@@ -31,7 +31,16 @@ function getDestinationFromConfig(source, target) {
     throw new Error("clio.toml was not found on the specified directory.");
   }
 
-  const buildConfig = packageConfig.getPackageConfig().build;
+  const config = packageConfig.getPackageConfig();
+  const buildConfig = config.build;
+
+  // TODO: Clean up assignment
```
Comment (pouya-eghbali) [inline cli/commands/build.js]:
```javascript
const buildTarget = target || buildConfig.build in config.target ?
  config.target[buildConfig.build].target : buildConfig.target
```
or

```javascript
const { target: configTarget } = config.target[buildConfig.build] || {}
const buildTarget = target || buildConfigTarget || buildConfig.target
```
we should find a clean way to replace this try / catch

Comment (garritfra) [reply cli/commands/build.js]: Ah, thanks for the suggestions!

Comment (garritfra) [reply cli/commands/build.js]: They both do not work, error is still thrown

Comment (pouya-eghbali) [reply cli/commands/build.js]: Righto, didn't test them, I'll try to find a way. I mean, one of those should work, with some modifications.

Comment (pouya-eghbali) [reply cli/commands/build.js]: oops, I had some `undefined` variables in my code, I edited it XD

Comment (garritfra) [reply cli/commands/build.js]: Fixed with a00a050

**Pull request merged**